scripts/tests/test_suites.sh

```diff
@@ -38,22 +42,103 @@ usage() {
     echo "  -s CASE_NAME: runs single test case name (e.g. Test_TC_OO_2_2"
     echo "                for Test_TC_OO_2_2.yaml) (by default, all are run)"
     echo "  -w COMMAND: prefix all instantiations with a command (e.g. valgrind) (default: '')"
+    echo "  -n Use linux netns to isolate app and tool executables"
+    echo "  -c execute a netns cleanup and exit"
     echo ""
     exit 0
 }
 
-# read shell arguments
-while getopts a:d:i:hs:w: flag; do
+declare app_run_prefix=""
+declare tool_run_prefix=""
+
+netns_setup() {
+    if [[ `id -u` -ne 0 ]]; then
+        echo 'Running as non-root user, restarting in unshare environment'
+        echo 'Executing: ' $0 $INPUT_ARGS
+        unshare --map-root-user -n -m $0 $INPUT_ARGS
+        exit 0
+    fi
+
+    echo 'Creating a separate mount'
+    mount --make-private /
+    mount -t tmpfs tmpfs /run
```

Comment:
This is not desirable unless the block on 56-60 happened. It will confuse things when running in docker, for example.

Also, if you run it twice in a row you'll probably get two stacked mounts, and so on.

Maybe add an option --mount-run for this and pass it on line 58?

Reply:
Done.